libs/design-system/src/navigation/NavMenu.tsx

```diff
@@ -9,32 +9,59 @@ interface INavMenuProps {
     label: string;
     link: string;
     testId?: string;
-    rightSide?: ReactNode;
+    rightSide?: ReactNode | { component: ReactNode; displayOnHover: boolean };
     condition?: boolean;
   }[];
 }
 
 export function NavMenu({ menuItems }: INavMenuProps) {
   const { classes } = useStyles();
+  const [isHovered, setIsHovered] = useState<string>(null);
 
   return (
     <div>
       {menuItems
         .filter(({ condition = true }) => condition)
-        .map(({ icon, link, label, testId, rightSide }) => (
-          <NavLink
-            key={link}
-            to={link}
-            className={({ isActive }) => (isActive ? classes.linkActive : classes.link)}
-            data-test-id={testId}
-          >
-            <Group spacing={10}>
-              <div className={classes.linkIcon}>{icon}</div>
-              <div className={classes.linkLabel}>{label}</div>
-            </Group>
-            <div>{rightSide}</div>
-          </NavLink>
-        ))}
+        .map(({ icon, link, label, testId, rightSide }) => {
+          return (
+            <NavLink
+              key={link}
+              to={link}
+              className={({ isActive }) => (isActive ? classes.linkActive : classes.link)}
+              data-test-id={testId}
+              onMouseEnter={() => setIsHovered(link)}
+              onMouseLeave={() => setIsHovered(null)}
+            >
+              <Group spacing={10}>
+                <div className={classes.linkIcon}>{icon}</div>
+                <div className={classes.linkLabel}>{label}</div>
+              </Group>
+              {rightSideSection({ rightSide: rightSide, isHover: isHovered === link })}
+            </NavLink>
+          );
+        })}
     </div>
   );
 }
+
+const rightSideSection = ({
```

Comment:
I'm not completely satisfied with this solution, and I'm uncertain if it's the most effective approach. 
The reason it's placed here is because we want to show the VisibilityOff Icon only when the navigation link is hovered. Additionally, the component's appearance was quite intense, so I looked for a way to create a smoother transition. This is the best method I could come up with.

Reply:
It will be easy to just use the native CSS display: block display none style
but it wont be easy to animate those transitions

Also i will suggest you to change the hover calculation

Create a NavLink component
and have the hover state inside that  Navlink

```jsx
const NavLinkComponent = (props) => {
const [isHover, setIsHover] = useState(false)

return <div onMouseEnter onMouseLeave>
<MantineNavlink  {...props}/>
{rightSideSection}
</div>

}


menuItems.map(item => <NavlinkComponent  ...item/>
```